tests/conftest.py

```diff
@@ -83,3 +90,208 @@ def setup_dynamodb():
             "ingest_table": ingest_table,
             "algorithm_table": algorithm_table,
         }
+
+
+def create_dependency_api_event(
+    source: str,
+    data_type: str,
+    descriptor="sci",
+    dep_type: str = "DOWNSTREAM",
+    relationship: str = "HARD",
+    start_date: Optional[str] = None,
+    end_date: Optional[str] = None,
+    version: Optional[str] = None,
+    trigger_source: Optional[str] = None,
+):
+    """Create event dictionaries for tests."""
+    event = {
+        "queryStringParameters": {
+            "dependency_type": dep_type,
+            "relationship": relationship,
+            "data_source": source,
+            "data_type": data_type,
+            "descriptor": descriptor,
+        }
+    }
+    optional_params = {
+        "start_date": start_date,
+        "end_date": end_date,
+        "version": version,
+        "trigger_source": trigger_source,
+    }
+    for param, value in optional_params.items():
+        if value:
+            if isinstance(value, list):
+                single_val = value[0]
+            else:
+                single_val = value
+            event["queryStringParameters"][param] = single_val
```

Comment:
What would a list of items mean, and is it OK to just take the first item and drop the rest?
```suggestion
        if value is not None:
            event["queryStringParameters"][param] = value
```

Reply:
I added the list functionality because I wanted to create a dependency api event directly from a url when mocking the call to the dependency lambda handler. I do that by calling urllib.parse.parse_qs which returns a dictionary of lists for each param. The url should only ever contain one item per param.

    parsed_url = parse.urlparse(url)
    params = parse.parse_qs(parsed_url.query) 

This is used in test_batch_starter.urlopen_side_effect() function